### ISSUE TITLE
Post tools may not send PMs to users

### ISSUE THREAD

**Issue by yuliu** [mybb/mybb#3922]
I'm not very sure how this happens. Here's a configuration for a post tool (default settings for not mentioned options):
- **Show Confirmation Page** -> `Yes` (may not relevant)
- **Split posts?** -> _an active forum (type 'Forum') other than "split to the same forum"_
- **Split thread prefix** -> _some thread prefix_ (may not relevant)
- **Send Private Message->Message** -> `Hi this is a test message....`
- **Send Private Message->Subject** -> `hello`

What I've done:
- Browse into a thread and tick several posts, and run this post tool, confirm.

At the user's end, whose posts get executed by the post tool, sometimes there will be a PM with exactly the same settings of subject & message but sometimes there's no PM received.

**Comment by yuliu** [comment]
OK, looks like the PM will only be sent to the thread's author but not the post's author(s). Is this the proper of sending PMs in a **post tool**?

**Comment by Sama34** [comment]
If the post was splited off the original thread then the post author should probably be notified and you shouldn't be able to execute post moderation tools on the first post.

**Comment by yuliu** [comment]
The fact seems to be only the thread's author will be notified but not the post's author unless they're the same user.

And the post tool can be and should be able to be applied to the first reply in a thread.

**Comment by Sama34** [comment]
Yes, I understand. I think thread tools should be executed only to the first post and the first post author should be the one notified. Post tools, on the other hand, should be executable only on replies (not the first post) and every post author should be notified.

Why is the thread author notified if a post is "unapproved" but not the post author? Why should you be able to split the first post off the thread? That makes not sense.

I would need more input to see if my approach is appropiate @mybb/developers

**Comment by yuliu** [comment]
My thoughts on sending PMs in post tools:
- Current 1.8's way isn't appropriate, which is a wrong design and should be corrected. The PMs in post tools should **at least be sent to authors of the posts** that are being moderated/operated.
- For future 1.9 version, there could be further improvement. **Respective options of _Yes_/_No_ for sending PMs to the post(s)' author(s) and to the thread's author** can be implemented. In some circumstance, yes, the thread's author might also need notifications.

On operations to the first post in a thread in post tools:
- As far as I know, standard moderation tools in the front-end allows multiple operations on the first post in a thread, including splitting. Ideally the first post in a thread should represent the thread, but things may not go well that the thread goes in wrong direction and the first post must be removed, and now splitting it from the thread seems to be the only way to do so. Anyway, posts other than the first post can be split, which results in two similar threads with different `tid`s.
- That means, operations to the first post should not be restricted.
